content/doc/book/system-administration/reverse-proxy-configuration-haproxy.adoc

```diff
@@ -45,27 +45,29 @@ defaults
    timeout server 60s
 
 frontend http-in
-   bind *:80
-   mode http
-   acl prefixed-with-jenkins  path_beg /jenkins/
-   acl host-is-jenkins-example   hdr(host) eq jenkins.example.com
-   use_backend jenkins if host-is-jenkins-example prefixed-with-jenkins
+  log stdout format raw local0 debug #for additional logging
+  bind *:80
+  mode http
+  acl prefixed-with-jenkins  path_beg /jenkins
+  http-request redirect code 301 prefix /jenkins unless prefixed-with-jenkins #Use http-request redirect prefix to add /jenkins prefix to the URL's location to ensure jenkins base url is working properly.
+  use_backend jenkins if prefixed-with-jenkins
 
 backend jenkins
-   server jenkins1 127.0.0.1:8080
-   mode http
-   reqrep ^([^\ :]*)\ /(.*) \1\ /\2
-   acl response-is-redirect res.hdr(Location) -m found
-   # Must combine following two lines into a SINGLE LINE for HAProxy
-   rspirep ^Location:\ (http|https)://127.0.0.1:8080/jenkins/(.*)
-           Location:\ \1://jenkins.example.com/jenkins/\2 if response-is-redirect
+  log stdout format raw local0 debug 
+  mode http
+  server jenkins1 127.0.0.1:8080 check
+  http-request replace-path /jenkins(/)?(.*) /\2
+  http-request set-header X-Forwarded-Port %[dst_port]
+  http-request add-header X-Forwarded-Proto https if { ssl_fc }
+  http-request set-header X-Forwarded-Host %[req.hdr(Host)]
+
 ----
 
 This assumes Jenkins is running locally on port 8080.
 
-This assumes that you are using the /jenkins/ context path for both the
+This assumes that you are using the http://localhost/jenkins[/jenkins] context path for both the
 site exposed from HAProxy, and Jenkins itself.
-If this is not the case, you will need to adjust the configuration.
+If this is not the case, you will need to adjust the configuration. Read HAProxy documentation on Traffic Routing by https://www.haproxy.com/documentation/hapee/latest/traffic-routing/[clicking here] for more information.
```

Comment:
```suggestion
If this is not the case, you will need to adjust the configuration.
Refer to the HAProxy documentation on link:https://www.haproxy.com/documentation/hapee/latest/traffic-routing/[Traffic Routing] for more information.
```
Small suggestion to place each sentence on its own line, update link for some syntax and suggestion on format.